File: smash-ranks-android/app/src/main/java/com/garpr/android/models/MatchesBundle.java
package com.garpr.android.models;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.Nullable;

import com.garpr.android.misc.ParcelableUtils;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class MatchesBundle implements Parcelable {

    @SerializedName("player")
    private AbsPlayer mPlayer;

    @Nullable
    @SerializedName("matches")
    private ArrayList<Match> mMatches;

    @SerializedName("losses")
    private int mLosses;

    @SerializedName("wins")
    private int mWins;


    public int getLosses() {
        return mLosses;
    }

    @Nullable
    public ArrayList<Match> getMatches() {
        return mMatches;
    }

    public AbsPlayer getPlayer() {
        return mPlayer;
    }

    public int getWins() {
        return mWins;
    }

    public boolean hasMatches() {
        return mMatches != null && !mMatches.isEmpty();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(final Parcel dest, final int flags) {
        ParcelableUtils.writeAbsPlayer(mPlayer, dest, flags);
        dest.writeTypedList(mMatches);
        dest.writeInt(mLosses);
        dest.writeInt(mWins);
    }

    public static final Creator<MatchesBundle> CREATOR = new Creator<MatchesBundle>() {
        @Override
        public MatchesBundle createFromParcel(final Parcel source) {
            final MatchesBundle mb = new MatchesBundle();
            mb.mPlayer = ParcelableUtils.readAbsPlayer(source);
            mb.mMatches = source.createTypedArrayList(Match.CREATOR);
            mb.mLosses = source.readInt();
            mb.mWins = source.readInt();
            return mb;
        }

        @Override
        public MatchesBundle[] newArray(final int size) {
            return new MatchesBundle[size];
        }
    };

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/models/MatchesBundle.kt
package com.garpr.android.models

import android.os.Parcel
import android.os.Parcelable
import com.garpr.android.extensions.createParcel
import com.garpr.android.extensions.readAbsPlayer
import com.garpr.android.extensions.writeAbsPlayer
import com.google.gson.annotations.SerializedName

data class MatchesBundle(
        @SerializedName("player") val player: AbsPlayer,
        @SerializedName("losses") val losses: Int = 0,
        @SerializedName("wins") val wins: Int = 0,
        @SerializedName("matches") val matches: List<Match>? = null
) : Parcelable {

    companion object {
        @JvmField
        val CREATOR = createParcel { MatchesBundle(it.readAbsPlayer(), it.readInt(),
                it.readInt(), it.createTypedArrayList(Match.CREATOR)) }
    }

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeAbsPlayer(player, flags)
        dest.writeInt(losses)
        dest.writeInt(wins)
        dest.writeTypedList(matches)
    }

}
